### ISSUE TITLE
useLinkComponent undefined

### ISSUE THREAD

**Issue by Siic19** [Shopify/polaris#295]
**This is the first time I'm opening an issue in my short career as a dev!**

Importing useLinkComponent and console logging it returns undefined.

```
import {
  useLinkComponent
} from '@shopify/polaris';

console.log(useLinkComponent);
```

It stops working at version 2.0.0-beta.9 and above, but works on all versions prior.

**Comment by AndrewMusgrave** [comment]
Hi @Siic19 thanks for the issue. With the introduction of the [AppProvider](https://polaris-v2.shopify.com/components/structure/app-provider#navigation) we've changed the way this behaves. The app provider takes a optional `linkComponent` prop that accepts the type `ReactComponent`, which will be a custom component for all polaris components to use

Here is an example of it's usage.
```jsx
class ProviderLinkExample extends React.Component {
  render() {
    const CustomLinkComponent = ({children, url, ...rest}) => {
      return (
        <a
          href={url}
          onClick={() => console.log('Custom link clicked')}
          {...rest}
        >
          {children}
        </a>
      );
    };

    return (
      <AppProvider linkComponent={CustomLinkComponent}>
        <Page
          breadcrumbs={[{content: 'Products', url: '#'}]}
          title="Jar With Lock-Lid"
          primaryAction={{content: 'Save', disabled: true}}
          secondaryActions={[
            {content: 'Duplicate', url: '#'},
            {content: 'View on your store', url: '#'},
          ]}
        >
          <p>Page content</p>
        </Page>
      </AppProvider>
    );
  }
}
```

Let me know if this answers your question!

**Comment by AndrewMusgrave** [comment]
@Siic19 The page component will work inside or outside an embedded app. The pages have different layouts, which is why providing `shopOrigin` and `apiKey` change the layout.

There was a bug in the way we handled `linkComponent` but a fix is in progress. We're hoping to include the fix in the next beta release. Regardless I'll keep you posted!

**Comment by dfmcphee** [comment]
This should be resolved as of `v2.0.0-rc.1`.
